### DISCUSSION TITLE
Machine parsable output for flux diff ks

### DISCUSSION THREAD

**Discussion by tun0** [fluxcd/flux2#3740] [Proposals]
We've been working on integrating `flux diff ks` into our CI pipelines. Currently we parse the "raw" output of `flux diff ks`. It'd make things much easier if it'd support alternative outputs like JSON, YAML, etc.

Primary reason to parse the output instead of just dumping it as-is: collapsing the actual changes. So at first glance you only get to see which resources changes, and after expanding such a resource, you can see the actual changes.

**Comment by souleb** [comment]
I think that 's a good idea. 

Our printer code is here: https://github.com/fluxcd/flux2/blob/d4cc45f55e74b4517823b31836d3c5671c2be826/pkg/printers/dyff.go#L39..

We could provide a new interface to do the parsing, but it would be better to contribute that code upstream to `[github.com/homeport/dyff](https://pkg.go.dev/github.com/homeport/dyff)`.

**Reply by tun0** [reply]
https://github.com/homeport/dyff/issues/13

5y old 😔
